examples/plot_afq_api.py

```diff
@@ -11,74 +11,162 @@
+# Stanford's Vista Lab
+#
+#   see https://purl.stanford.edu/ng782rw8378 for details on dataset.
+#
+# The data for the first subject and first session are downloaded locally
+# (by default into the users home directory) under:
+#
+#   ``.dipy/stanford_hardi/``
+#
+# Anatomical data (``anat``) and Diffusion-weighted imaging data (``dwi``) are
+# then extracted, formatted to be BIDS compliant, and placed in the AFQ
+# data directory (by default in the users home directory) under:
+#
+#   ``AFQ_data/stanford_hardi/``
+#
+# This data represents the required preprocessed diffusion data necessary for
+# intializing the AFQ object (which we will do next)
 
 afd.organize_stanford_data()
-base_dir = op.join(op.expanduser('~'), 'AFQ_data', 'stanford_hardi')
 
 ##########################################################################
 # Initialize an AFQ object:
 # ------------------------
+#
+# Creates an AFQ object, that encapsulates tractometry. This object can be
+# used to manage the entire AFQ pipeline, including:
+#
+# - Tractography
+# - Registration
+# - Segmentation
+# - Cleaning
+# - Profiling
+# - Visualization
+#
+# In this example we will load the subjects session data from the previous step
+# using the default AFQ parameters.
+#
+# .. note::
+#
+#    The first time intializing the AFQ object will download necessary
+#    waypoint regions of interest (ROIs) templates into AFQ data directory:
+#
+# - Human corpus callosum templates: ``AFQ_data/callosum_templates/``
+#
+#   see https://digital.lib.washington.edu/researchworks/handle/1773/34926
+#
+# - Tract probability maps: ``AFQ_data/templates/``
+#
+#   see https://figshare.com/articles/Tract_probability_maps_for_automated_fiber_quantification/6270434
+#
+# These waypoints ROIs will used to identify the desired white matter tracts.
+#
+# This will also create an output folder for the corresponding AFQ derivatives
+# in the AFQ data directory: ``AFQ_data/stanford_hardi/derivatives/afq/``
+#
+# To initialize this object we will pass in the path location to our BIDS
+# compliant data.
+#
+# .. note::
+#
+#    As noted above, the Stanford HARDI data contains anatomical and
+#    diffusion weighted imaging (dwi) data. In this example, we are intrested
+#    in the vistasoft dwi. For our dataset the `dmriprep` is optional, but
+#    we have included it to make the initialization more explicit.
+#
+# .. note::
+#
+#    We will also be using plotly to geneate an interactive visualization.
+#    So we will specify plotly as the visualization backend.
 
 myafq = api.AFQ(bids_path=op.join(afd.afq_home,
                                   'stanford_hardi'),
                 dmriprep='vistasoft',
-                viz_backend="plotly")
+                viz_backend='plotly')
 
 ##########################################################################
-# Reading in DTI FA
+# Reading in DTI FA (Diffusion Tensor Imaging Fractional Anisotropy)
```

Comment:
I think that you need to extend the `----` below to the length of the text (otherwise Sphinx raises an annoying warning).